src/websockethandler.cpp

```diff
@@ -15,6 +15,57 @@ namespace {
 Logger logger(LOG_MAIN, "WebSocketHandler");
 }  // namespace
 
+ExponentialBackoffStrategy::ExponentialBackoffStrategy() {
+  MVPN_COUNT_CTOR(ExponentialBackoffStrategy);
+
+  connect(&m_retryTimer, &QTimer::timeout, this,
+          &ExponentialBackoffStrategy::onNextAttemptTimeout);
```

Comment:
connect the timeout signal with the exeucteNextAtempt directly: `&ExponentialBackoffStrategy::executeNextAttempt)`. In this way you can remove `onNextattemptTimeout`